group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -1302,6 +1312,9 @@ public void run() {
                     if (!future.isDone()) {
                         operationTimeout = new OperationTimeout(tp, this, writeTimeout.toMillis());
                         timer.add(operationTimeout);
+
+                        // Only update when the records were appended to the local log.
+                        writeTimestamp = time.milliseconds();
```

Comment:
What's the reasoning for only updating the metric if records were appended? The event may wait in the purgatory even without writing records.

Reply:
I renamed this to `deferredEventQueuedTimestamp`.

It's only marked when the event was added to the deferred queue